bionic/flow.py

```diff
@@ -844,46 +844,81 @@ def to_builder(self):
 
         return FlowBuilder._from_state(self._state)
 
-    def get(self, name, fmt=None):
+    def get(self, name, collection=None, fmt=None, mode=object):
         """
         Computes the value(s) associated with an entity.
 
-        If the entity has multiple values, the ``fmt`` parameter indicates
-        how to handle them.  It can have any of the following values:
+        If the entity has multiple values, the ``collection`` parameter
+        indicates how to handle them.  It can have any of the following values:
 
-        * ``object``: return a single value or throw an exception
+        * ``None``: return a single value or throw an exception
         * ``list`` or ``'list'``: return a list of values
         * ``set`` or ``'set'``: return a set of values
         * ``pandas.Series`` or ``'series'``: return a series whose index is
           the root cases distinguishing the different values
 
+        The user can specify the type of object (implicitly specifying in-memory vs. persisted data)
+        to return in the collection using the ``mode`` parameter.  It can have any of the
+        following values:
+        * ``object`` for a value in-memory
+        * ``FileCopier`` for a wrapper for a path to the persisted file for the computed entity
+        * ``Path`` or ``'path'`` for a path to persisted file
+        * ``'filename'`` for a string representing a path to a persisted file
+
         Parameters
         ----------
 
         name: String
             The name of a entity.
-        fmt: String or type, optional, default is ``object``
+
+        collection: String or type, optional, default is ``None``
             The data structure to use if the entity has multiple values.
 
+        fmt: String or type, optional, default is ``None``
+            The data structure to use if the entity has multiple values.  Deprecated in favor of
+            ``collection`` and will be removed in future release.
+
+        mode: String or type, optional, default is ``object``
+            The type of object to return in the collection.
+
         Returns
         -------
 
         The value of the entity, or a collection containing its values.
         """
 
         result_group = self._deriver.derive(name)
-        if fmt is None or fmt is object:
-            if len(result_group) == 0:
+        if mode is object:
```

Comment:
should we handle the string version of this as well, e.g. `if mode is object or mode == 'object'`?